### DISCUSSION TITLE
Expose api for deno like node-swc

### DISCUSSION THREAD

**Discussion by lidessen** [swc-project/swc#3097] [Ideas]
### Describe the feature

Can you provide an official library for `Deno` just like `node-swc` please!

**Comment by kwonoj** [comment]
Moving this into discussions as this is a feature request to be discussed.

**Comment by kwonoj** [comment]
I _think_ this maybe feasible once we have native esm support.

**Reply by lidessen** [reply]
I found something from `napi-rs`, they are planning to support `Deno FFI`

[[napi] Support Deno FFI #900](https://github.com/napi-rs/napi-rs/issues/900#issue-1069086819)

**Reply by kwonoj** [reply]
ah, forgot ffi part. So there's blocking upstream issue.

**Reply by lidessen** [reply]
Since SWC already use `napi-rs` to provide node support, I think this will help

**Reply by lidessen** [reply]
> ah, forgot ffi part. So there's blocking upstream issue.

So we need to wait for upstream updates 😂

**Reply by XiNiHa** [reply]
Should we actually use `napi-rs` for Deno FFI? Since the NodeJS interface is modularized, it sounds like it's also possible to build a separated Deno FFI crate, with tools like [`deno_bindgen`](https://github.com/littledivy/deno_bindgen) for example (haven't used it tho)

**Reply by kwonoj** [reply]
maybe, but I guess that'll require different compilation path as well as maintaining 2 binding entrypaths as well.

Either way, I think it's ok to leave current feature request and see how upstream / ecosystem evolves. I do not expect we'll have this in a week or month.

**Comment by footearth** [comment]
```js
import swc from "https://esm.sh/@swc/core"

console.log(typeof swc)
```

```
error: Import 'https://cdn.esm.sh/v66/@swc/core-darwin-arm64@1.2.147/deno/core-darwin-arm64.js' failed: 500 Internal Server Error
    at https://cdn.esm.sh/v66/@swc/core@1.2.147/deno/core.js:2:1344
```

```
/* esm.sh - error */
throw new Error("[esm.sh] " + "esbuild: No loader is configured for \".node\" files: esm-build-30f025a464f7d55699c27d1d7259e9d27a9365b9-0c569f82/node_modules/@swc/core-darwin-arm64/swc.darwin-arm64.node");
export default null;
```

**Reply by kdy1** [reply]
Use `@swc/wasm` or `@swc/wasm-web` instead.
